src/features/metrics/aggregate/index.js

```diff
@@ -1,17 +1,26 @@
-import { getConfigurationValue } from "../../../common/config/config";
+import { getConfigurationValue, getRuntime } from "../../../common/config/config";
 import { registerHandler } from "../../../common/event-emitter/register-handler";
 import { HarvestScheduler } from "../../../common/harvest/harvest-scheduler";
 import { AggregateBase } from "../../utils/aggregate-base";
-import { FEATURE_NAME } from "../constants";
+import { FEATURE_NAME, SUPPORTABILITY_METRIC, CUSTOM_METRIC, SUPPORTABILITY_METRIC_CHANNEL, CUSTOM_METRIC_CHANNEL } from "../constants";
 import { drain } from '../../../common/drain/drain'
-
+import { getFrameworks } from '../../../common/metrics/framework-detection'
+import { protocol } from '../../../common/url/protocol'
+import { getRules, validateRules } from '../../../common/util/obfuscate'
+import { VERSION } from '../../../common/constants/environment-variables'
+import { onDOMContentLoaded } from '../../../common/window/load'
+import { windowAddEventListener } from '../../../common/event-listener/event-listener-opts'
+import { isBrowserScope } from '../../../common/util/global-scope'
 export class Aggregate extends AggregateBase {
     static featureName = FEATURE_NAME
     constructor(agentIdentifier, aggregator) {
         super(agentIdentifier, aggregator, FEATURE_NAME)
 
-        registerHandler('storeMetric', (...args) => this.storeMetric(...args), this.featureName, this.ee)
-        registerHandler('storeEventMetrics', (...args) => this.storeEventMetrics(...args), this.featureName, this.ee)
+        this.singleChecks() // checks that are run only one time, at script load
+        this.eachSessionChecks()    // the start of every time user engages with page
+
+        registerHandler(SUPPORTABILITY_METRIC_CHANNEL, (...args) => this.storeSupportabilityMetrics(...args), this.featureName, this.ee)
+        registerHandler(CUSTOM_METRIC_CHANNEL, (...args) => this.storeEventMetrics(...args), this.featureName, this.ee)
```

Comment:
QQ should the registration of the handler be done before we make the checks? Thinking of a possible race condition where the checks fire before the handler is registered.

Also, these can be shortened to `registerHandler(SUPPORTABILITY_METRIC_CHANNEL, this.storeSupportabilityMetrics.bind(this), this.featureName, this.ee)` and `registerHandler(CUSTOM_METRIC_CHANNEL, this.storeEventMetrics.bind(this), this.featureName, this.ee)`

Reply:
Should have no effect in this case, since the calls in `eachSessionChecks` directly call the `this.storeSupportabilityMetrics()` method instead of passing a message to the EE.  I will reorder for consistency tho